tests/Doctrine/Tests/ORM/Query/QueryExpressionVisitorTest.php

```diff
@@ -39,7 +39,13 @@ public function testWalkComparison(CriteriaComparison $criteriaExpr, $queryExpr,
         }
     }
 
-    /** @psalm-return list<array{CriteriaComparison, QueryBuilder\Comparison|string, ?Parameter} */
+    /**
+     * @psalm-return list<array{
+     *                   0: CriteriaComparison,
+     *                   1: QueryBuilder\Comparison|QueryBuilder\Func|string,
+     *                   2?: Parameter,
```

Comment:
Is this different than the previous specification for this parameter? Previously it meant always present but possibly null, whereas here it would mean never null, but optional?

Reply:
Exactly. And if you look at the data provider you'll find that the declaration "never null, but optional" is more accurate.

Oh, and the closing `>` was missing, so the previous type declaration was invalid and thus ignored.